File: backend/src/main/java/com/placesync/service/InterviewService.java
package com.placesync.service;

import com.placesync.dto.request.InterviewRequest;
import com.placesync.dto.response.InterviewResponse;
import com.placesync.enums.InterviewStatus;

import java.util.List;

public interface InterviewService {

    InterviewResponse scheduleInterview(
            InterviewRequest request
    );

    InterviewResponse getInterviewById(Long id);

    List<InterviewResponse> getAllInterviews();

    List<InterviewResponse> getInterviewsByStatus(
            InterviewStatus status
    );

    InterviewResponse updateInterviewStatus(
            Long id,
            InterviewStatus status
    );

    void cancelInterview(Long id);
}

File: frontend/src/services/interviewService.js
import api from "./api";

export const getAllInterviews = async () => {
  const response = await api.get("/interviews");
  return response.data;
};

export const scheduleInterview = async (interviewData) => {
  const response = await api.post("/interviews", interviewData);
  return response.data;
};

export const updateInterviewStatus = async (id, status) => {
  const response = await api.put(
    `/interviews/${id}/status?status=${status}`
  );

  return response.data;
};

export const cancelInterview = async (id) => {
  await api.put(`/interviews/${id}/cancel`);
};